File: app/src/main/java/net/rmoreno/weatherapp/HourlyAdapter.java
package net.rmoreno.weatherapp;

import android.content.Context;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by Rafa on 12/10/15.
 */
public class HourlyAdapter extends RecyclerView.Adapter<HourlyAdapter.ViewHolder> {

    public class ViewHolder extends RecyclerView.ViewHolder{

        public TextView mTemperature;
        public TextView mTime;
        public CardView mCard;

        public ViewHolder(View v){
            super(v);

            mTemperature = (TextView) v.findViewById(R.id.temperature);
            mTime = (TextView) v.findViewById(R.id.time);
            mCard = (CardView) v.findViewById(R.id.hourlyCard);
        }
    }

    ArrayList<HourlyWeather> mDataset;
    Context mContext;

    public HourlyAdapter(Context context, ArrayList<HourlyWeather> hourly){
        mContext = context;
        mDataset = hourly;
    }

    @Override
    public HourlyAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.hourly_view, parent, false);

        ViewHolder holder = new ViewHolder(v);

        return holder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

        holder.mTemperature.setText(String.valueOf(mDataset.get(position).getTemp()));
        holder.mTime.setText(mDataset.get(position).getFormatedTime());

        if(position % 4 == 0){
            holder.mCard.setCardBackgroundColor(mContext.getResources().getColor(R.color.teal));
        }
        else if(position % 4 == 1){
            holder.mCard.setCardBackgroundColor(mContext.getResources().getColor(R.color.yellow));

        }
        else if(position % 4  == 2){
            holder.mCard.setCardBackgroundColor(mContext.getResources().getColor(R.color.orange));
        }
        else if(position % 4 == 3){
            holder.mCard.setCardBackgroundColor(mContext.getResources().getColor(R.color.red));
        }

    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }
}

File: app/src/main/java/net/rmoreno/weatherapp/adapters/HourlyAdapter.kt
package net.rmoreno.weatherapp.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import net.rmoreno.weatherapp.R
import net.rmoreno.weatherapp.models.HourlyWeather
import java.util.*

class HourlyAdapter(private var dataset: ArrayList<HourlyWeather>) : RecyclerView.Adapter<HourlyAdapter.ViewHolder>() {

    inner class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {

        var temperature: TextView = v.findViewById(R.id.temperature) as TextView
        var time: TextView = v.findViewById(R.id.time) as TextView

        var card: CardView = v.findViewById(R.id.hourlyCard) as CardView

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HourlyAdapter.ViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.hourly_view, parent, false)

        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.temperature.text = dataset[position].temp.toString()
        holder.time.text = dataset[position].formattedTime

        when {
            position % 4 == 0 -> holder.card.setCardBackgroundColor(holder.itemView.context.resources.getColor(R.color.teal))
            position % 4 == 1 -> holder.card.setCardBackgroundColor(holder.itemView.context.resources.getColor(R.color.yellow))
            position % 4 == 2 -> holder.card.setCardBackgroundColor(holder.itemView.context.resources.getColor(R.color.orange))
            position % 4 == 3 -> holder.card.setCardBackgroundColor(holder.itemView.context.resources.getColor(R.color.red))
        }
    }

    override fun getItemCount(): Int {
        return dataset.size
    }
}
